### DISCUSSION TITLE
System Landscape and relations

### DISCUSSION THREAD

**Discussion by mydeveloperplanet** [structurizr/ui#55] [Q&A] [answered]
I am investigating whether Structurizr can be applied at a customer's site. They have multiple applications which interact with each other. Every application is modeled using the C4 model (System Context Diagram). Let's assume application A interacts with application B. Application A has modeled the relationship with application B, but application B has 'forgotten' to model this. 

In order to find a solution for this problem, I have experimented with Structurizr on-premises. I have used the following example: https://github.com/structurizr/examples/tree/main/dsl/system-catalog

I have created separate workspaces for 'Order Service', 'Customer Service', 'Invoice Service' and for the 'System Landscape'.

I have some questions about this:
1. The System Landscape is generated, that is nice. However, I would like to click on the 'Order Service' in order that it would open the 'Order Service'-workspace. This is possible when the DSL is defined in the same workspace (between SystemContext and Containers f.e.), but can this be done in this scenario?
2. The 'Order Service' and 'Invoice Service' interact with the 'Customer Service'. The relationship is defined in the 'Order Service' and 'Invoice Service'. However, the relationship is not defined in the 'Customer Service' and therefore, it is not shown in the 'Customer Service'-workspace. Is it possible to have an error generated somewhere in order to be notified of this?

**Comment by simonbrowndotje** [comment, marked as answer]
I have a new version of that example at https://github.com/structurizr/examples/tree/main/enterprise ... 1 is implemented by the example, and 2 should be relatively straightforward via this approach too.

**Reply by mydeveloperplanet** [reply]
Thank you for your quick response:
1. correct, I managed to use that example and it works just fine.
2. I managed to implement some basis validation checks. So yes, it is possible.

**Reply by roland-part** [reply]
Hi guys, i was wondering the same thing, how to open another workspace from the system landscape - ended up on this thread. The referenced example links are not working any longer unfortunately - could you help me find the examples please? 
The closest example i could find is [this](https://github.com/structurizr/examples/blob/1223caa6ffe98276767fd008f3cdc8f1dacca01c/dsl/big-bank-plc/system-landscape/workspace.dsl#L9) - is it the one you talked about above? If so, am i right to assume i would need to use the static url with the id of the target workspace?
Thanks in advance

**Reply by mydeveloperplanet** [reply]
I made an example based on the above information: https://mydeveloperplanet.com/2024/04/03/architect-your-system-landscape-as-code/
